src/python/pants/backend/python/goals/lockfile.py

```diff
@@ -158,6 +225,112 @@ async def generate_lockfile(
         ),
     )
 
+
+@rule
+async def generate_updated_lockfile(
+    update_req: UpdatePythonLockfileRequest,
+    generate_lockfiles_subsystem: GenerateLockfilesSubsystem,
+    pex_lock_subsystem: PexLockSubsystem,
+    python_setup: PythonSetup,
+) -> ProcessResult:
+    req = update_req.gen_req
+    pip_args_setup = await _setup_pip_args_and_constraints_file(req.resolve_name)
+
+    for project in pex_lock_subsystem.project:
+        requirement = Requirement(project)
+        if requirement.name != project:
+            raise ValueError(
+                f"project {project} is not a bare name; specifier, markers must be specified in BUILD files"
+            )
+
+    # Assuming that any lockfile we are trying to update has a destination
+    # that is a file and not an embedded read only resource.  See
+    # pex_requirements.py
+    maybe_lockfile = await Get(
```

Comment:
Unsure if this is actually the idiomatic way to check if a file exists.

Reply:
(I don't know.)